velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -1035,17 +1049,50 @@ void assertResults(
     const std::string& duckDbSql,
     DuckDbQueryRunner& duckDbQueryRunner) {
   MaterializedRowMultiset actualRows;
+  TypePtr type = nullptr;
   for (const auto& vector : results) {
     auto rows = materialize(vector);
     std::copy(
         rows.begin(), rows.end(), std::inserter(actualRows, actualRows.end()));
   }
 
   auto expectedRows = duckDbQueryRunner.execute(duckDbSql, resultType);
-  assertEqualResults(
-      expectedRows,
-      actualRows,
-      kDuckDbTimestampWarning + "\nDuckDB query: " + duckDbSql);
+
+  // For logical types like DECIMAL, we need to create the TypePtr
+  // from the plan provided for using it with variant::toJson().
+  if (resultType->size() > 0) {
```

Comment:
I don't understand why we need this block of code. Can you explain?

Reply:
@majetideepak - The resultType was having size 0 in one testcase - `HashJoinTest.reclaimDuringInputProcessing`. 
<img width="1745" alt="Screenshot 2023-08-17 at 10 29 07 AM" src="https://github.com/facebookincubator/velox/assets/136469600/548fe2a8-9f04-4594-9b34-1f6ad87eccf9">

I have fixed to not use `if-else` by adding this statement and using this with `switch-case` so that it moves to default always. Let me know if this is what we expect here.
```
const auto typeKind = resultType->size() > 0 ? resultType->childAt(0)->kind() : TypeKind::ROW;
```

Reply:
`resultType` should always be present. 
Which is the test that fails?

Reply:
In file `velox/exec/tests/HashJoinTest.cpp` the test `reclaimDuringInputProcessing`.

Reply:
This test returns a valid type which is an empty ROW. Why is this a problem?
I see you are extracting decimal type from the first children of resultType. Why not pass the resultType as is?

Reply:
So without any child available for this row, the childAt(0) will throw error while trying to extract the decimal TypePtr.

In such situations the TypKind becomes Row, and the default case of the switch gets executed. If we check there, we pass the resultType directly for such cases where we couldn't find decimal type.

Reply:
With considering the changes we expect to do with toJson() via other PR, we should just pass the resultType directly. The rest should be handled by toJson. I will make the changes.

Reply:
Makes sense to handle this in toJson(). Thanks!

Reply:
Fixed.